src/main/java/ua/vedroid/cinema/dao/impl/UserDaoImpl.java

```diff
@@ -67,8 +67,11 @@ public User add(User user) {
     @Override
     public Optional<User> findByEmail(String email) {
         try (Session session = sessionFactory.openSession()) {
-            return session.createQuery("from User where email = :email", User.class)
-                    .setParameter("email", email).uniqueResultOptional();
+            return session.createQuery("from User u "
+                    + "left join fetch u.roles "
```

Comment:
let's use inner join here, because there shouldn't be a situation when user doesn't have roles